src/os.cpp

```diff
@@ -153,21 +174,71 @@ static int get_ttyfd(void) {
 }
 #endif // HAVE_TERMIOS
 
+/// \brief Signal raised whenever TTY size changes
+static void os_SIGWINCH_handler(int sig) {
+    (void) sig;
+    auto *s = get_state();
+    if (!s->initialized) {
+        return;
+    }
+    // It's not safe to do complex logic in signal handlers,
+    // therefore we will actually update the console size in the next get size request.
+    s->resize_pending = true;
+}
+
+bool os_update_tty_size(tty_state *s) {
+#ifdef HAVE_TTY
+#if defined(HAVE_TERMIOS) && defined(HAVE_IOCTL)
+    winsize ws{};
+    if (ioctl(STDIN_FILENO, TIOCGWINSZ, &ws) == 0) {
+        if (ws.ws_col >= 1 && ws.ws_row >= 1) {
+            s->cols = ws.ws_col;
+            s->rows = ws.ws_row;
+            return true;
+        }
+    } else {
+#ifdef DEBUG_OS
+        (void) fprintf(stderr, "os_update_tty_size(): ioctl TIOCGWINSZ failed\n");
+#endif
+    }
+
+#else
+    // TODO(edubart): what to do on Windows and MacOS?
```

Comment:
Are we going to figure this out before merge?

Reply:
There are still some `TODO(edubart)`, the PR is WIP, they will all be fixed before merging.